pkg/apis/management.cattle.io/v3/authz_types.go

```diff
@@ -106,26 +106,76 @@ type GlobalRoleBinding struct {
-	ClusterCreatorDefault bool                `json:"clusterCreatorDefault,omitempty" norman:"required"`
-	ProjectCreatorDefault bool                `json:"projectCreatorDefault,omitempty" norman:"required"`
-	Context               string              `json:"context" norman:"type=string,options=project|cluster"`
-	RoleTemplateNames     []string            `json:"roleTemplateNames,omitempty" norman:"type=array[reference[roleTemplate]]"`
-	Administrative        bool                `json:"administrative,omitempty"`
+	// DisplayName is the human-readable name used for this resource.
+	// +kubebuilder:validation:Required
+	DisplayName string `json:"displayName,omitempty" norman:"required"`
+
+	// Description holds any text desired to better describes the resource.
+	// +optional
+	Description string `json:"description"`
+
+	// Rules holds all the PolicyRules for this RoleTemplate
+	// +optional
+	Rules []rbacv1.PolicyRule `json:"rules,omitempty"`
+
+	// Builtin if true specifies that this RoleTemplate was created by Rancher and is immutable.
+	// Default to false.
+	// +optional
+	Builtin bool `json:"builtin" norman:"nocreate,noupdate"`
+
+	// External if true specifies that rules for this RoleTemplate should be gathered from a ClusterRole with the matching name
+	// If set to true the Rules on the template will not be evaluated.
+	// External's value is only evaluated if the RoleTemplate's context is set to "cluster"
+	// Default to false.
+	// +optional
+	External bool `json:"external"`
+
+	// Hidden if true informs the Rancher UI not to display this RoleTemplate.
+	// Default to false.
+	// +optional
+	Hidden bool `json:"hidden"`
+
+	// Locked if true, new bindings will not be able to use this RoleTemplate.
+	// Default to false.
+	// +optional
+	Locked bool `json:"locked,omitempty" norman:"type=boolean"`
+
+	// ClusterCreatorDefault if true, a binding with this RoleTemplate will be created for a users when they create a new cluster.
+	// ClusterCreatorDefault is only evaluated if the context of the RoleTemplate is set to cluster.
+	// Default to false.
+	// +optional
+	ClusterCreatorDefault bool `json:"clusterCreatorDefault,omitempty" norman:"required"`
+
+	// ProjectCreatorDefault if true, a binding with this RoleTemplate will be created for a users when they create a new project.
+	// ProjectCreatorDefault is only evaluated if the context of the RoleTemplate is set to project.
+	// Default to false.
+	// +optional
+	ProjectCreatorDefault bool `json:"projectCreatorDefault,omitempty" norman:"required"`
+
+	// Context describes if the roleTemplate applies to clusters or projects.
+	// Valid values are "project" and "cluster".
+	// +kubebuilder:validation:Enum={"project","cluster"}
+	// +kubebuilder:validation:Required
+	Context string `json:"context" norman:"type=string,options=project|cluster"`
+
+	// RoleTemplateNames list of RoleTemplate names that this RoleTemplate will inherit.
```

Comment:
Nit: Might be good to explain what "inherit" means in this context (i.e. rules given from the inherited role will be applied by this role).

Reply:
can do 👍🏾